robotoff/prediction/category/neural/keras_category_classifier_3_0/text_utils.py

```diff
@@ -0,0 +1,1442 @@
+    (0xFF07, "'"),
+    (0x2010, "-"),
+    (0x2011, "-"),
+    (0x2012, "-"),
+    (0x2013, "-"),
+    (0x2014, "-"),
+    (0x207B, "-"),
+    (0x208B, "-"),
+    (0xFF0D, "-"),
+    (0x2045, "["),
+    (0x2772, "["),
+    (0xFF3B, "["),
+    (0x2046, "]"),
+    (0x2773, "]"),
+    (0xFF3D, "]"),
+    (0x207D, "("),
+    (0x208D, "("),
+    (0x2768, "("),
+    (0x276A, "("),
+    (0xFF08, "("),
+    (0x2E28, "(("),
+    (0x207E, ")"),
+    (0x208E, ")"),
+    (0x2769, ")"),
+    (0x276B, ")"),
+    (0xFF09, ")"),
+    (0x2E29, "))"),
+    (0x276C, "<"),
+    (0x2770, "<"),
+    (0xFF1C, "<"),
+    (0x276D, ">"),
+    (0x2771, ">"),
+    (0xFF1E, ">"),
+    (0x2774, "{"),
+    (0xFF5B, "{"),
+    (0x2775, "}"),
+    (0xFF5D, "}"),
+    (0x207A, "+"),
+    (0x208A, "+"),
+    (0xFF0B, "+"),
+    (0x207C, "="),
+    (0x208C, "="),
+    (0xFF1D, "="),
+    (0xFF01, "!"),
+    (0x203C, "!!"),
+    (0x2049, "!?"),
+    (0xFF03, "#"),
+    (0xFF04, "$"),
+    (0x2052, "%"),
+    (0xFF05, "%"),
+    (0xFF06, "&"),
+    (0x204E, "*"),
+    (0xFF0A, "*"),
+    (0xFF0C, ","),
+    (0xFF0E, "."),
+    (0x2044, "/"),
+    (0xFF0F, "/"),
+    (0xFF1A, ":"),
+    (0x204F, ";"),
+    (0xFF1B, ";"),
+    (0xFF1F, "?"),
+    (0x2047, "??"),
+    (0x2048, "?!"),
+    (0xFF20, "@"),
+    (0xFF3C, "\\"),
+    (0x2038, "^"),
+    (0xFF3E, "^"),
+    (0xFF3F, "_"),
+    (0x2053, "~"),
+    (0xFF5E, "~"),
+]
+
+translate_table = codepoint_to_self + codepoint_to_replacement
+
+
+def fold(string: str, replacement: str = "") -> str:
+    """Fold string to ASCII.
+
+    Unmapped characters should be replaced with empty string by default, or other
+    replacement if provided.
+
+    All astral plane characters are always removed, even if a replacement is
+    provided.
+    """
+
+    if string is None:
+        return ""
+
+    try:
+        # If string contains only ASCII characters, return it.
+        string.encode("ascii")
+        return string
+    except UnicodeEncodeError:
+        pass
+
+    if replacement:
+        t = defaultdict(lambda: replacement, translate_table)
+    else:
+        t = defaultdict(lambda: None, translate_table)  # type: ignore
+
+    return string.translate(t)
+
+
+CONSECUTIVE_SPACES_REGEX = re.compile(r" {2,}")
+
+
+def strip_consecutive_spaces(text: str) -> str:
+    """Convert a sequence of 2+ spaces into a single space."""
+    return CONSECUTIVE_SPACES_REGEX.sub(" ", text)
+
+
+def get_tag(text: str) -> str:
+    """Return a tag from a text.
+
+    In Open Food Facts, tags are obtained from free text by performing the
+    following:
+    - lowercasing
+    - accent removal
+    - replacement of punctuation by either a comma ("-") or nothing, depending
+    on the punctuation
+    """
+    text = fold(text)
+    text = (
+        text.lower()
+        .replace(" & ", "-")
+        .replace(" ", "-")
+        .replace("'", "-")
+        .replace(".", "-")
+        .replace("!", "")
+        .replace("?", "")
+    )
```

Comment:
Can't we use unicodedata and the punctuation class to remove other punctuation, like maybe the spanish ¿